src/adapter/src/catalog/storage.rs

```diff
@@ -121,12 +123,6 @@ async fn migrate(
                 },
                 IdAllocValue { next_id: 1 },
             )?;
-            txn.id_allocator.insert(
-                IdAllocKey {
-                    name: SYSTEM_ROLE_ID_ALLOC_KEY.into(),
```

Comment:
I think you need to add a migration that removes this key from the allocator stash too.

Reply:
I added this to the existing migration on line 414:
```Rust
            // Delete the system role id allocator. All system role ids will need
            // to be hard-coded going forward.
            txn.id_allocator
                .delete(|id_alloc_key, _| id_alloc_key.name == "system_role");
```